`containerlab deploy` panics when sharing namespace with an existing container created by kind

Hi, I found this problem after upgrading containerlab from 0.51.3 to 0.52.0.

## What

When I try to create a network with `kind`, `containerlab` panics with the following outputs.

```console
kind create cluster --name test  
Creating cluster "test" ...
 ✓ Ensuring node image (kindest/node:v1.26.3) 🖼 
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-test"
You can now use your cluster with:

kubectl cluster-info --context kind-test

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/
sudo bin/containerlab deploy -t test.yaml deploy
INFO[0000] Containerlab v0.0.0 started                  
INFO[0000] Parsing & checking topology file: test.yaml  
INFO[0000] Creating lab directory: /home/terassyi/workspace/containerlab/clab-test 
WARN[0000] node test-control-plane refrenced in namespace sharing not found, considering it an external dependency. 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1a4aaca]

goroutine 1 [running]:
github.com/srl-labs/containerlab/clab/dependency_manager.(*DependencyNode).AddDepender(0x0, {0x29e66ef, 0x6}, 0xc0005a4b10, {0x29e66ef, 0x6})
        /home/terassyi/workspace/containerlab/clab/dependency_manager/dependency_node.go:133 +0x8a
github.com/srl-labs/containerlab/clab.(*CLab).createNamespaceSharingDependency(0xc0005a7550)
        /home/terassyi/workspace/containerlab/clab/clab.go:445 +0x1a6
github.com/srl-labs/containerlab/clab.(*CLab).CreateNodes(0xc0005a7550, {0x3131250, 0xc00089c320}, 0x2, 0x0)
        /home/terassyi/workspace/containerlab/clab/clab.go:387 +0x112
github.com/srl-labs/containerlab/cmd.deployFn(0x4783240?, {0x29e421a?, 0x4?, 0x29e421e?})
        /home/terassyi/workspace/containerlab/cmd/deploy.go:219 +0x10cf
github.com/spf13/cobra.(*Command).execute(0x4783240, {0xc0007a8120, 0x3, 0x3})
        /home/terassyi/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0x4780ce0)
        /home/terassyi/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        /home/terassyi/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/srl-labs/containerlab/cmd.Execute()
        /home/terassyi/workspace/containerlab/cmd/root.go:49 +0x1a
main.main()
        /home/terassyi/workspace/containerlab/main.go:10 +0xf
```

My configuration is here.
```yaml
name: test
topology:
  kinds:
    linux:
      cmd: bash
  nodes:
    node0:
      kind: linux
      image: frrouting/frr:v8.4.0
      exec:
        - ip a
    control-plane:
      kind: linux
      image: nicolaka/netshoot:latest
      network-mode: container:test-control-plane
      exec:
        - ip a
  links:
    - endpoints: ["node0:net0", "control-plane:net0"]
```

## Why

I think this change(https://github.com/srl-labs/containerlab/pull/1902) is related to this problem.

This panic is caused by a nil pointer dereference in the following line.
https://github.com/srl-labs/containerlab/blob/622a2c831b4870be6bbc60de6d191217d78f7bda/clab/dependency_manager/dependency_node.go#L133

When running it with my configuration, I also found that `referencedNode` is nil at line 440.

https://github.com/srl-labs/containerlab/blob/de03337aa9742c5949d69eda2011099a39af5192/clab/clab.go#L440-L445

---

https://github.com/srl-labs/containerlab/pull/1902 seems to change the configuration format.
So, I'm not sure if this is a bug or a specification change.

--- a/clab/clab.go
+++ b/clab/clab.go
@@ -439,7 +439,7 @@ func (c *CLab) createNamespaceSharingDependency() {
 		referenceNodeName := netModeArr[1]
 		referenceNode, err := c.dependencyManager.GetNode(referenceNodeName)
 		if err != nil {
-			log.Warnf("node %s refrenced in namespace sharing not found, considering it an external dependency.", referenceNodeName)
+			log.Warnf("node %s referenced in namespace sharing not found, considering it an external dependency.", referenceNodeName)
 		}
 
 		referenceNode.AddDepender(types.WaitForCreate, n, types.WaitForCreate)

--- a/clab/clab.go
+++ b/clab/clab.go
@@ -439,7 +439,8 @@ func (c *CLab) createNamespaceSharingDependency() {
 		referenceNodeName := netModeArr[1]
 		referenceNode, err := c.dependencyManager.GetNode(referenceNodeName)
 		if err != nil {
-			log.Warnf("node %s referenced in namespace sharing not found, considering it an external dependency.", referenceNodeName)
+			log.Warnf("node %s referenced in namespace sharing not found in topology definition, considering it an external dependency.", referenceNodeName)
+			continue
 		}
 
 		referenceNode.AddDepender(types.WaitForCreate, n, types.WaitForCreate)

--- a/tests/06-ext-container/02-shared-namespace-ext.clab.yaml
+++ b/tests/06-ext-container/02-shared-namespace-ext.clab.yaml
@@ -1,0 +1,14 @@
+name: shared-namespace-ext
+prefix: ""
+
+topology:
+  kinds:
+    linux:
+      cmd: sh
+  nodes:
+    ext-node:
+      kind: linux
+      image: alpine
+      exec:
+        - ip l add dev d1 type dummy
+        - ip a add dev d1 128.66.0.1/32

--- a/tests/06-ext-container/02-shared-namespace.clab.yaml
+++ b/tests/06-ext-container/02-shared-namespace.clab.yaml
@@ -1,0 +1,20 @@
+name: shared-namespace
+
+topology:
+  kinds:
+    linux:
+      cmd: sh
+  nodes:
+    node0:
+      kind: linux
+      image: alpine
+      exec:
+        - ip a
+    node1:
+      kind: linux
+      image: alpine
+      network-mode: container:ext-node
+      exec:
+        - ip a
+  links:
+    - endpoints: ["node0:net0", "node1:net0"]

--- a/tests/06-ext-container/02-shared-namespace.robot
+++ b/tests/06-ext-container/02-shared-namespace.robot
@@ -1,0 +1,72 @@
+*** Settings ***
+Library             OperatingSystem
+Library             Process
+Resource            ../common.robot
+
+Suite Setup         Setup
+Suite Teardown      Run Keyword    Cleanup
+
+
+*** Variables ***
+${lab-name}         06-shared-namespace
+${lab-file-name-1}    02-shared-namespace-ext.clab.yaml
+${lab-file-name-2}    02-shared-namespace.clab.yaml
+${runtime}          docker
+
+
+*** Test Cases ***
+Deploy ${lab-name}-ext lab
+    Log    ${CURDIR}
+    ${output} =    Run Process
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} deploy -d -t ${CURDIR}/${lab-file-name-1}
+    ...    shell=True
+    Log    ${output.stdout}
+    Log    ${output.stderr}
+    Should Be Equal As Integers    ${output.rc}    0
+
+Deploy ${lab-name} lab
+    Log    ${CURDIR}
+    ${output} =    Run Process
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} deploy -d -t ${CURDIR}/${lab-file-name-2}
+    ...    shell=True
+    ...    timeout=30s
+    Log    ${output.stdout}
+    Log    ${output.stderr}
+    Should Be Equal As Integers    ${output.rc}    0
+
+Verify ip on ext-node
+    ${rc}    ${output} =    Run And Return Rc And Output
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} exec --label clab-node-name\=ext-node --cmd "ip address show dev d1"
+    Log    ${output}
+    Should Be Equal As Integers    ${rc}    0
+    Should Contain    ${output}    128.66.0.1/32
+
+Verify links in node0
+    ${rc}    ${output} =    Run And Return Rc And Output
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} exec --label clab-node-name\=node0 --cmd "ip link show dev net0"
+    Log    ${output}
+    Should Be Equal As Integers    ${rc}    0
+    Should Contain    ${output}    state UP
+
+Verify ext-node defined interface is present for node1
+    ${rc}    ${output} =    Run And Return Rc And Output
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} exec --label clab-node-name\=node1 --cmd "ip address show dev d1"
+    Log    ${output}
+    Should Be Equal As Integers    ${rc}    0
+    Should Contain    ${output}    128.66.0.1/32
+
+Verify topo defined link in node1
+    ${rc}    ${output} =    Run And Return Rc And Output
+    ...    sudo -E ${CLAB_BIN} --runtime ${runtime} exec --label clab-node-name\=node1 --cmd "ip link show dev net0"
+    Log    ${output}
+    Should Be Equal As Integers    ${rc}    0
+    Should Contain    ${output}    state UP
+
+
+*** Keywords ***
+Setup
+    Cleanup
+
+Cleanup
+    Run    sudo -E ${CLAB_BIN} --runtime ${runtime} destroy -t ${CURDIR}/${lab-file-name-1} --cleanup
+    Run    sudo -E ${CLAB_BIN} --runtime ${runtime} destroy -t ${CURDIR}/${lab-file-name-2} --cleanup

